app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -69,6 +68,8 @@ def index
         instance_ids = instances.pluck(:library_instance_id).reject(&:blank?)
         instance_ids += instances.where(:library_instance_id => nil)
         repositories = Repository.where(:id => instance_ids)
+      elsif params[:library] && params[:content_view_version_id] && params[:organization_id]
+        repositories = repositories.where(:environment_id => @organization.library.id, :content_view_version_id => params[:content_view_version_id])
```

Comment:
This logic will fall down if the content view version is not currently in the library. I think you want something more like:

```
repositories = Repository.where(:id => repositories.map(&:library_instance_id))
```